marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/QueryStepRunner.java

```diff
@@ -373,33 +348,38 @@ private RunStepResponse runHarmonizer(RunStepResponse runStepResponse, Collectio
         Vector<String> errorMessages = new Vector<>();
 
         // The client used here doesn't matter, given that a QueryBatcher is going to be constructed based on an
-        // Iterator. It's the construction of the FlowResource that matters, as that makes use of destinationDatabase
-        // to control where outputted documents are written to.
+        // Iterator. It's the step options that determine where documents are written to.
         dataMovementManager = hubClient.getStagingClient().newDataMovementManager();
 
+        final StepRunnerService stepRunnerService = StepRunnerService.on(hubClient.getStagingClient());
```

Comment:
Should we use batch.getClient() to initialize the StepRunnerService so that request will be sent to all nodes?
Silvano had previously filed a bug for this https://project.marklogic.com/jira/browse/DHFPROD-1172

Reply:
Ah - I'll add that back in and add a comment to explain why it's being done.

Reply:
I made the change - presumably, the DatabaseClient will not have "database" set on it, since the DC used to construct the DataMovementManager doesn't either. If "database" is set, the DataServices-generated interface will complain.